### ISSUE TITLE
why check partition twice when creating metaPartition on the metanode?

### ISSUE THREAD

**Issue by biuboombiuboom** [chubaofs/chubaofs#706]
/metanode/manager.go  line 368
```go
func (m *metadataManager) createPartition(request *proto.CreateMetaPartitionRequest) (err error) {
	// check partitions  once       
	if _, err = m.getPartition(request.PartitionID); err == nil {
		err = errors.NewErrorf("create partition id=%d is exsited!", request.PartitionID)
		return
	}

	err = nil
	partitionId := fmt.Sprintf("%d", request.PartitionID)

	mpc := &MetaPartitionConfig{
		....
	}
	mpc.AfterStop = func() {
		m.detachPartition(request.PartitionID)
	}
  
	oldMp, err := m.getPartition(request.PartitionID)
	if err == nil {
		if err = oldMp.IsEquareCreateMetaPartitionRequst(request); err != nil {
			err = errors.NewErrorf("[createPartition]->%s", err.Error())
			return
		}
		return nil
	}

	partition := NewMetaPartition(mpc, m)
	if err = partition.PersistMetadata(); err != nil {
		err = errors.NewErrorf("[createPartition]->%s", err.Error())
		return
	}
	if err = m.attachPartition(request.PartitionID, partition); err != nil {
		os.RemoveAll(mpc.RootDir)
		err = errors.NewErrorf("[createPartition]->%s", err.Error())
		return
	}
	return
}
```

once
```go
if _, err = m.getPartition(request.PartitionID); err == nil {
		err = errors.NewErrorf("create partition id=%d is exsited!", request.PartitionID)
		return
}
```

again

```go
oldMp, err := m.getPartition(request.PartitionID)
	if err == nil {
		if err = oldMp.IsEquareCreateMetaPartitionRequst(request); err != nil {
			err = errors.NewErrorf("[createPartition]->%s", err.Error())
			return
		}
		return nil
}
```

**Comment by biuboombiuboom** [comment]
when create datapartition， it was checked only once, and use the lock
```
manager.partitionMutex.Lock()
	defer manager.partitionMutex.Unlock()
	dpCfg := &dataPartitionCfg{
		PartitionID:   request.PartitionId,
		VolName:       request.VolumeId,
		Peers:         request.Members,
		Hosts:         request.Hosts,
		RaftStore:     manager.raftStore,
		NodeID:        manager.nodeID,
		ClusterID:     manager.clusterID,
		PartitionSize: request.PartitionSize,
	}
	dp = manager.partitions[dpCfg.PartitionID]
	if dp != nil {
		if err = dp.IsEquareCreateDataPartitionRequst(request); err != nil {
			return nil, err
		}
		return
	}
	disk := manager.minPartitionCnt()
	if disk == nil {
		return nil, ErrNoSpaceToCreatePartition
	}
	if dp, err = CreateDataPartition(dpCfg, disk, request); err != nil {
		return
	}
	manager.partitions[dp.partitionID] = dp

	return
```

**Comment by mervinkid** [comment]
@biuboombiuboom 

Thank you for your review.
You found a duplicate and useless process.
MetaNode performed repeated and useless checks during the creation of MetaPartition.

Would you like to submit a pull request to solve this problem.

**Comment by mervinkid** [comment]
@biuboombiuboom 

Your pull request (#711 ) have been approved and merged into master branch.
Thanks for your contribution.
